File: backend/src/container/dto/create-container.dto.ts
import { IsString, IsOptional, IsNumber } from 'class-validator';

export class CreateContainerDto {
  @IsString()
  @IsOptional()
  registry?: string;

  @IsString()
  imageName: string;

  @IsString()
  @IsOptional()
  tag?: string;

  @IsNumber()
  @IsOptional()
  scanIntervalMinutes?: number;

  @IsString()
  @IsOptional()
  scanCron?: string;
}

File: backend-java/src/main/java/com/asmolabs/zanshin/container/dto/CreateContainerDto.java
package com.asmolabs.zanshin.container.dto;

import jakarta.validation.constraints.NotBlank;

public record CreateContainerDto(
    String registry,
    @NotBlank(message = "Image name is required")
    String imageName,
    String tag,
    Integer scanIntervalMinutes,
    String scanCron
) {}
